consensus/src/step_votes_reg.rs

```diff
@@ -49,19 +47,10 @@ impl CertificateInfo {
     pub(crate) fn add_sv(
         &mut self,
         iter: u8,
-        hash: [u8; 32],
         sv: StepVotes,
         svt: SvType,
         quorum_reached: bool,
     ) -> bool {
-        if let Some(h) = self.hash {
```

Comment:
As stated, reaching a quorum in non-Emergency mode depends on the `aggregator` state but not `sv_registry` state , so I don't expect that the issue will be resolved with this change.

PS. However, I do agree the patch beneficial for failed_certificates correctness.

Reply:
Then I could remove the "resolve" for the issue. But keep in mind that with this PR I'm not able to replicate the issue anymore, so in someway this resolves that

Reply:
> Then I could remove the "resolve" for the issue. But keep in mind that with this PR I'm not able to replicate the issue anymore, so in someway this resolves that

we should figure out WHY this PR solves that issue...

Reply:
It's up to you. I'd rather investigate why and what this is resolving. If the problem is due to a race condition, you might be unable to reproduce it consistently.